lib/fluent/plugin/out_dynatrace.rb

```diff
@@ -173,6 +176,20 @@ def failure_message(res)
 
         "failed to request #{uri} (#{res_summary})"
       end
+
+      #############################################
+
+      private
+
+      def parse_and_validate_uri(uri_string)
+        raise Fluent::ConfigError, 'active_gate_url is empty' if uri_string.empty?
+
+        uri = URI.parse(uri_string)
+        raise Fluent::ConfigError, 'active_gate_url scheme must be http or https' unless
+          %w[http https].include?(uri.scheme)
```

Comment:
Added in [c80eed3](https://github.com/dynatrace-oss/fluent-plugin-dynatrace/pull/34/commits/c80eed318f0e9d585bf4d264f6ad600adc8ea74b). 

By checking if the host is set, we can check if the URL contains an authority, which requires the existence of the leading `//` before the authority part. Otherwise `URI.parse()` would throw.

Protocol-relative URLs are already caught by checking for `http` or `https`, as with a Protocol-relative URL the `scheme` property would be empty after `URI.parse()`. This would trigger the `active_gate_url scheme must be http or https`-ConfigError

Reply:
ah i see this is the scheme already anyway i was just off-base here